### PR TITLE
Srch_rm_more Merge - Various Search modifications

### PR DESCRIPTION
**Pull request by michaeta**
- Removed user input on search
- Changed search CLI: Search CLI now uses the textwrap package to display summaries.
- Removed logging silence hack from search
- Changed search tests to comply with modifications

### CODE REVIEW

**Comment by michaeta** [comment]
Not sure how all those Merges got in there, they don't show up under my branch... They are just me merging in various changes into my repo.

File: pip2/commands/search.py [hunk 1]
```diff
@@ -8,56 +8,31 @@
...
     installed = pip2.commands.freeze.freeze()
 
     for project in projects:
         results[project.name] = dict()
-        # temp for future ref.
-        # Sometimes a release can't be found or
-        # version number can't be rationalized
+        # if all releases for a project have irrational versions then the
+        # releases list will be empty
         try:
```
Comment (njwilson) [inline pip2/commands/search.py]: The try/except should be replaced with an if/else. I think you should be able to do something like `if len(project.releases) > 0:`

**Comment by njwilson** [comment]
Looks pretty good. Here's a couple more things that you could change:
- Copy/paste a new search result into README.rst that shows the new line wrapping.
- You have an `except (UnicodeDecodeError, UnicodeEncodeError, UnicodeError):` in a couple places. You can replace these with just an `except UnicodeError:` since [the others are subclasses.](http://docs.python.org/py3k/library/exceptions.html#exception-hierarchy)

I'm not sure where those merge commits came from either. I may investigate, or I may wait until we meet again on Tuesday to figure it out.

**Comment by njwilson** [comment]
It looks like your develop branch got out of sync with the upstream develop branch. I'm not sure how that happened, but we can fix it again. If you're feeling brave, try following these steps when you don't have any don't have any local changes (any files that you've modified that you haven't committed). Otherwise we can try this on Tuesday.

Make sure you are in your develop branch:

```
$ git checkout develop
```

Create a copy of your develop branch called develop-old for backup. This command won't switch you to the other branch, so you will still be in your develop branch.

```
$ git branch develop-old
```

Reset your local develop branch so it exactly matches the upstream develop branch from osupython/pip2:

```
$ git reset --hard a46e1b7a4d250763e50995ea78233e5d469be3cf
```

Force push your local develop branch to michaeta/pip2. The --force will remove whatever was there before and replace it with your local develop branch.

```
$ git push --force origin develop
```

Your local develop branch and your develop branch on michaeta/pip2 should now match the develop branch on osupython/pip2 (unless there have been new upstream commits since I wrote these instructions, but that's fine).

Now you can rebase your srch_rm_more on top of the fixed develop branch. This should get rid of all those extra merge commits.

```
$ git checkout srch_rm_more
$ git rebase develop
```

If you do a `git log`, you should see your new commits on top of commit a46e1b7a4d250763e50995ea78233e5d469be3cf.

If everything looks good, force push your local srch_rm_more branch to michaeta/pip2 to completely replace the code in your pull request with your new "fixed" code:

```
$ git push --force origin srch_rm_more
```

From now on, the only time you need to modify your develop branch is when you are pulling in the latest upstream changes. If you add the --ff-only (fast-forward only) flag to the merge or pull command, it will refuse to do the merge/pull if it would create a merge commit. If you only modify your develop branch to pull in upstream changes, it should never need to do a merge commit. For example, you could update your develop branch like this:

```
$ git checkout develop
$ git fetch upstream
$ git merge --ff-only upstream/develop
```

If the command fails with the --ff-only flag, something is wrong with your local develop branch again.

**Comment by njwilson** [comment]
Oh, there's a silly bug in the search code when the terminal width is too small (probably smaller than the width you have allocated for package names). The `wrap` method in `textwrap` raises a ValueError when you give it a negative number:

```
ValueError: invalid width -8 (must be > 0)
```

After your code gets the terminal width, it should probably verify that it's >= some minimum value. If not, set it to the minimum value.

File: pip2/commands/search.py (hunk 1 above)
Comment (merwok) [inline pip2/commands/search.py]: Or take advantage of boolean evaluation in Python: if project.releases

**Comment by njwilson** [comment]
Alright, we fixed your develop branch, got rid of the excess merge requests, and did a push --force to update the pull request.

Merged!

**Pull request merged**